plugins/modules/vmware_vm_storage_policy_info.py

```diff
@@ -135,9 +135,15 @@ def get_storage_policy_info(self):
                 subprofiles = profile.constraints.subProfiles
                 temp_sub_profiles = []
                 for subprofile in subprofiles:
+                    rule_set_info = self.show_capabilities(subprofile.capability)
+                    # if a storage policy set tag base placement rules, the tags are set into the value.
+                    # https://github.com/ansible-collections/community.vmware/issues/742
+                    for _rule_set_info in rule_set_info:
+                        if isinstance(_rule_set_info['value'], pbm.capability.types.DiscreteSet):
+                            _rule_set_info['value'] = _rule_set_info['value'].values
```

Comment:
@sky-joker Do you think we need to assign this variable back to `rule_set_info`? This looks no-op to me.

Reply:
Pardon me, if I do not understand this block of code.

Reply:
Thank you @Akasurde for reviewing.

> Do you think we need to assign this variable back to rule_set_info? This looks no-op to me.

May I confirm your intention?
Does it mean that should we define the tags as a new property into the return value is better, not rule_set_info?

Reply:
Should I remove the tags from the result?

Reply:
Sorry, I am unable to understand this for loop. I think at line 146, `rule_set_info` should be `'rule_set_info': _rule_set_info,`, otherwise, there is no point of above for loop. Let me know if my understanding is correct.

Reply:
Thanks for your response.

If my understanding is correctly, the `_rule_set_info[value]` is the same memory address to `rule_set_info[value]`.  
Please see the below example.

```python
#!/usr/bin/env python

import json

rule_set_info = [
    {
        "name": "example1",
        "value": 1
    },
    {
        "name": "example2",
        "value": 2
    }
]

print("before")
print(json.dumps(rule_set_info, indent=2))

for _rule_set_info in rule_set_info:
    if _rule_set_info["name"] == "example1":
        _rule_set_info["value"] = 3

print("---------")
print("after")
print(json.dumps(rule_set_info, indent=2))
```

The above example code is that the value will be overwritten to 3 if the key is example1.

```
$ ./example.py
before
[
  {
    "name": "example1",
    "value": 1
  },
  {
    "name": "example2",
    "value": 2
  }
]
---------
after
[
  {
    "name": "example1",
    "value": 3
  },
  {
    "name": "example2",
    "value": 2
  }
]
```

In this way, I implemented overwriting the value in the module, too.  
Does my answer make sense?

Reply:
oh got it. Thanks for the explanation.